### PR TITLE
add update restart policy

### PR DESCRIPTION
**Pull request by ningmingxiao**

### CODE REVIEW

File: cmd/nerdctl/run_restart.go [hunk 1]
```diff
@@ -65,3 +65,10 @@ func updateContainerStoppedLabel(ctx context.Context, container containerd.Conta
 	})
 	return container.Update(ctx, containerd.UpdateContainerOpts(opt))
 }
+
+func updateContainerResatrtPolicyLabel(ctx context.Context, container containerd.Container, policyLabel string) error {
+	opt := containerd.WithAdditionalContainerLabels(map[string]string{
+		restart.PolicyLabel: policyLabel,
+	})
+	return container.Update(ctx, containerd.UpdateContainerOpts(opt))
```
Comment (junnplus) [inline cmd/nerdctl/run_restart.go:73]:
```suggestion
	return container.Update(ctx, containerd.UpdateContainerOpts(restart.WithPolicy(policy)))
```

File: cmd/nerdctl/run_restart.go [hunk 2]
```diff
@@ -65,3 +65,10 @@ func updateContainerStoppedLabel(ctx context.Context, container containerd.Conta
 	})
 	return container.Update(ctx, containerd.UpdateContainerOpts(opt))
 }
+
+func updateContainerResatrtPolicyLabel(ctx context.Context, container containerd.Container, policyLabel string) error {
```
Comment (junnplus) [inline cmd/nerdctl/run_restart.go:69]: We should check the capabilities of the restart plugin, https://github.com/containerd/nerdctl/blob/master/cmd/nerdctl/run_restart.go#L36-L50

**Review by junnplus — CHANGES_REQUESTED** [review]

File: cmd/nerdctl/run_restart.go (hunk 2 above)
Comment (ningmingxiao) [reply cmd/nerdctl/run_restart.go]: thanks

**Comment by junnplus** [comment]
Can we have an integration test?

**Comment by ningmingxiao** [comment]
> Can we have an integration test?

done

File: cmd/nerdctl/update.go [hunk 3]
```diff
@@ -325,6 +329,20 @@ func updateContainer(ctx context.Context, client *containerd.Client, id string,
...
+func updateContainerRestartPolicy(ctx context.Context, client *containerd.Client, container containerd.Container, cmd *cobra.Command) (retErr error) {
+	policyLabel, err := cmd.Flags().GetString("restart")
+	if err != nil {
+		return err
+	}
+	if cmd.Flags().Changed("restart") && policyLabel != "" {
+		if err = updateContainerResatrtPolicyLabel(ctx, client, container, policyLabel); err != nil {
+			log.G(ctx).WithError(err).Errorf("Failed to update policyLabel %s for container %q", policyLabel, container.ID())
+		}
+		return err
```
Comment (junnplus) [inline cmd/nerdctl/update.go:341]: nit: unnecessary

**Comment by junnplus** [comment]
Please rebase master to fix CI, thanks.

File: cmd/nerdctl/update.go (hunk 3 above)
Comment (ningmingxiao) [reply cmd/nerdctl/update.go:341]: I can‘t understand which is unnecessary

Comment (junnplus) [reply cmd/nerdctl/update.go:341]: `return err`

**Comment by ningmingxiao** [comment]
already rabase @junnplus

**Comment by AkihiroSuda** [comment]
CI failing

File: cmd/nerdctl/run_restart.go [hunk 4]
```diff
@@ -65,3 +48,38 @@ func updateContainerStoppedLabel(ctx context.Context, container containerd.Conta
 	})
 	return container.Update(ctx, containerd.UpdateContainerOpts(opt))
 }
+
+func updateContainerResatrtPolicyLabel(ctx context.Context, client *containerd.Client, container containerd.Container, policyLabel string) error {
```
Comment (manugupt1) [inline cmd/nerdctl/run_restart.go:52]: Nit: typo restart policy

**Comment by junnplus** [comment]
Seems like CI always fails, but neither is relevant

**Comment by djdongjin** [comment]
Hi @ningmingxiao, any update on this PR? seems the `run` side change is implemented now and the `update` side is not.

thanks for your work. :)

**Comment by djdongjin** [comment]
Probably I can give a hand so the PR can be merged :) cc @AkihiroSuda

**Comment by djdongjin** [comment]
Close as #1682 is merged. #1682 carries the changes here and the original commit in this PR is kept in #1682. Thanks.

**Pull request closed without merging**